### PR TITLE
hwdef: added CUAV Pixhawk6X-IE support

### PR DESCRIPTION
**Pull request by mxiaogit**
CUAV Pixhawk6X / CUAV Pixhawk6X IE hardware

### CODE REVIEW

File: libraries/AP_Baro/AP_Baro_ICP201XX.cpp [hunk 1]
```diff
@@ -0,0 +1,486 @@
...
+    uint8_t fifo_data[96] {0};
+    uint8_t fifo_packets = 0;
+    int32_t data_temp = 0;
+    int32_t data_press = 0;
+    *pressure = 0;
+    *temperature = 0;
+
+    if (read_reg(REG_FIFO_FILL, &fifo_packets)) {
+        fifo_packets  = (uint8_t)(fifo_packets & 0x1F);
+        if (fifo_packets > 0 && fifo_packets <= 16 && read_reg(REG_FIFO_BASE, fifo_data, fifo_packets * 2 * 3)) {
```
Comment (tridge) [inline libraries/AP_Baro/AP_Baro_ICP201XX.cpp:257]: can fifo_packets be over 16? If so it looks like we will stop reading

Comment (mxiaogit) [reply libraries/AP_Baro/AP_Baro_ICP201XX.cpp:257]: ccording to the datasheet, the fifo_packets maximum value is 16.

Comment (tridge) [reply libraries/AP_Baro/AP_Baro_ICP201XX.cpp:257]: the datasheet shows this:
image
the problem is that if it ever does give one of these reserved values then the driver will never continue. I think you either need to call flush_fifo() here if fifo_packets > 16

Comment (mxiaogit) [reply libraries/AP_Baro/AP_Baro_ICP201XX.cpp:257]: yes

**Comment by tridge** [comment]
this PR is replaced by #21217, closing

**Pull request closed without merging**